src/taipy/_run.py

```diff
@@ -49,8 +59,15 @@ def _run(*apps: t.List[t.Union[Gui, Rest, Core]], **kwargs) -> t.Optional[t.Unio
         return gui.run(**kwargs)
     else:
         app = rest or gui
+        assert app is not None  # for pyright typing
```

Comment:
> And this would be due to a pyright bug/limitation, right? No impact on the runtime, fortunately...

It's more that when you have this logic:
```python
    if not rest and not gui:
        return None
    if gui and rest:
        gui._set_flask(rest._app)
        return gui.run(**kwargs)
    else:
        app = rest or gui
        assert app is not None  # Avoid pyright typing error
        return app.run(**kwargs)
```
when you get to the `else`, taipy can't infer that `rest` or `gui` must not be `None`.  Wouldn't say it's a bug.  It's actually a feature that `pyright` does quite well in general, but I reported the type narrowing issue here:  https://github.com/microsoft/pyright/issues/4643

`mypy` has the same issue, although I'm not sure why it didn't complain in your case, but if does have the same problem as what I put in the `pyright` issue.